enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -826,7 +842,20 @@ func (p *PebbleCache) updateAtime(key filestore.PebbleKey) error {
 	if !olderThanThreshold(atime, p.atimeUpdateThreshold) {
 		return nil
 	}
-	md.LastAccessUsec = p.clock.Now().UnixMicro()
+
+	newAtime := p.clock.Now()
+
+	// If this is a GCS object, update the custom time and record the new
+	// custom time.
+	if gcsMetadata := md.GetStorageMetadata().GetGcsMetadata(); gcsMetadata != nil {
+		if err := p.fileStorer.UpdateBlobAtime(p.env.GetServerContext(), gcsMetadata, newAtime); err != nil {
+			log.Errorf("Error updating GCS custom time: %s", err)
```

Comment:
instead of error, maybe it's worth to add some metrics.

Reply:
Done.